fastapi/exceptions.py

```diff
@@ -54,7 +54,7 @@ def __init__(
             ),
         ] = None,
         headers: Annotated[
-            Optional[Dict[str, str]],
+            Optional[Mapping[str, str]],
```

Comment:
`Mapping` (like `Dict`) doesn't inherently support duplicate keys. This change wouldn't solve the problem. To handle this scenario properly, I suggest replacing 

`Optional[Mapping[str, str]]` 

with 

`Optional[List[Tuple[str, str]]]`

Let me know your thoughts.

Reply:
`Mapping` was chosen because starlette already uses it in every place where a generic header type is required:
- `starlette.datastructures.Headers` inherits from `Mapping[str, str]` and receives an optional `headers: Mapping[str, str]` in its `__init__`: https://github.com/encode/starlette/blob/8a99adfb5839b37efee33f2f49c1ba27a954fcbd/starlette/datastructures.py#L490-L500
- `starlette.responses.Response` also receives `Mapping[str, str]` in its `__init__`: https://github.com/encode/starlette/blob/8a99adfb5839b37efee33f2f49c1ba27a954fcbd/starlette/responses.py#L36

Also, `Mapping` is commonly used as a base class for such containers with non-unique keys:
- httpx: https://github.com/encode/httpx/blob/26d48e0634e6ee9cdc0533996db289ce4b430177/httpx/_types.py#L44-L50 (also features `Sequence[Tuple[str, str]]` though)
- requests: https://github.com/python/typeshed/blob/b0768596d2f7b2c12aa61d8a9714325284410b4b/stubs/requests/requests/api.pyi#L9
- django: https://github.com/django/django/blob/5e998d717f7b4220a1728bd49b66ca0162e2a6cb/django/http/request.py#L468 (`CaseInsensitiveMapping` inherits from `Mapping`)

I think being compatible with starlette is an important reason in and of itself.